### PR TITLE
WIP - Wqp 1294

### PR DESCRIPTION
**Pull request by mbucknell**

### CODE REVIEW

**Comment by mbucknell** [comment]
This PR covers the ETL from mysql to nwis schema, as well as the transformation steps for OrgData, MonitoringLocation, Result, ResultDetectionQuantitationLimit. I have removed the old .sql and .sh files that implemented the ETLs covered by this PR.

**Comment by mbucknell** [comment]
This is no longer a WIP PR despite the title.

File: src/main/java/gov/acwi/wqp/etl/result/ResultProcessor.java [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+		} else {
+			switch (remarkCd) {
+				case "U":
+				case "M":
+				case "N":
+				case "<":
+				case ">":
+					return null;
+				default:
+					return resultVa;
```
Comment (dsteinich) [inline src/main/java/gov/acwi/wqp/etl/result/ResultProcessor.java]: Not sure if it is ok to not call out the values in the original query here. ie is a P possible? if so, the original code wouldn't return a value...

Comment (mbucknell) [reply src/main/java/gov/acwi/wqp/etl/result/ResultProcessor.java]: Original Query is ```nvl(fxd.fxd_tx,
           case
             when r.remark_cd in ('U', 'M', 'N', '<', '>') then null
             when r.remark_cd in ('R', 'V', 'S', 'E', 'A') or r.remark_cd is null then r.result_va
             else r.result_va
           end)```
I read this that fi fxd.fxd_tx is null r.result_va is returned unless remark_cd is in ('U', 'M', 'N', '<', '>') . Am I reading this wrong?

File: src/main/java/gov/acwi/wqp/etl/nwis/TaskNwisStationLocalUpsert.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+				+ "round(sitefile.dec_lat_va , 7) latitude, round(sitefile.dec_long_va, 7) longitude, "
+				+ "sitefile.huc_cd huc, "
+				+ "case when sitefile.dec_long_va is not null and sitefile.dec_lat_va is not null then "
+				+ "st_SetSrid(st_MakePoint(dec_long_va, dec_lat_va), 4269) else null end geom "
+				+ "from sitefile on conflict (station_id) do "
+				+ "update set latitude = excluded.latitude, "
+				+ "longitude = excluded.longitude, "
+				+ "huc = excluded.huc,"
+				+ "calculated_huc_12 = null, "
+				+ "geom = excluded.geom where excluded.latitude != nwis_station_local.latitude or "
```
[the remark this answers is not in the document]
Comment (mbucknell) [reply src/main/java/gov/acwi/wqp/etl/nwis/TaskNwisStationLocalUpsert.java]: Once I added test data, I realized that I didn't want the where clause on line 36. Not sure if this still is the correct thing to do.

File: src/main/java/gov/acwi/wqp/etl/orgData/TransformOrgData.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	@Autowired
+	@Qualifier("buildOrgDataIndexesFlow")
+	private Flow buildOrgDataIndexesFlow;
+
+	@Bean
+	public JdbcCursorItemReader<NwisDistrictCdsByHost> nwisOrgReader() {
+		return new JdbcCursorItemReaderBuilder<NwisDistrictCdsByHost>()
+				.dataSource(dataSourceNwis)
+				.name("organizationReader")
+				.sql("select * from nwis_district_cds_by_host")
```
[the remark this answers is not in the document]
Comment (mbucknell) [reply src/main/java/gov/acwi/wqp/etl/orgData/TransformOrgData.java]: Added it back in.

File: etl-nwis.iml [hunk 4]
```diff
@@ -0,0 +1,92 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dsteinich) [inline etl-nwis.iml]: Looks like the .gitignore didn't work for this file.

File: src/main/java/gov/acwi/wqp/etl/result/ResultProcessor.java (hunk 1 above)
Comment (dsteinich) [reply src/main/java/gov/acwi/wqp/etl/result/ResultProcessor.java]: no, you are correct.

File: src/main/java/gov/acwi/wqp/etl/nwis/TaskNwisStationLocalUpsert.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dsteinich) [reply src/main/java/gov/acwi/wqp/etl/nwis/TaskNwisStationLocalUpsert.java]: The original intent to to only set the huc12 to null on existing rows IF the lat, long, or huc have changed. This is what triggers a recalculation of the HUC. no need to recalculate unless values have changed OR we reload the huc shape files (In which case we will truncate the nwis_station_local table to force them ALL to recalc).

[the remark this answers is not in the document]
Comment (mbucknell) [reply src/main/java/gov/acwi/wqp/etl/nwis/TaskNwisStationLocalUpsert.java]: is distinct from is the replacement for lvvl

File: src/main/resources/sql/nwisStationLocal.sql [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+		end geom
+	from sitefile
+on conflict (station_id) do
+    update set
+        latitude = excluded.latitude,
+        longitude = excluded.longitude,
+        huc = excluded.huc,
+        calculated_huc_12 = null,
+        geom = excluded.geom
+    where excluded.latitude is distinct from nwis_station_local.latitude or
```
Comment (mbucknell) [inline src/main/resources/sql/nwisStationLocal.sql]: Here is the postgres version of the upsert with the is distinct from.

Comment (dsteinich) [reply src/main/resources/sql/nwisStationLocal.sql]: I'd like to see some data to test this.

**Comment by mbucknell** [comment]
Added specific test data for nwis_station_local update that includes checking to make sure the calculated_huc12 field is nulled when appropriate.

**Pull request merged**